crates/ruff/src/rules/pylint/rules/repeated_isinstance_calls.rs

```diff
@@ -38,16 +40,31 @@ use crate::checkers::ast::Checker;
 /// - [Python documentation](https://docs.python.org/3/library/functions.html#isinstance)
 #[violation]
 pub struct RepeatedIsinstanceCalls {
-    obj: String,
-    types: Vec<String>,
+    expr: String,
 }
 
-impl Violation for RepeatedIsinstanceCalls {
+impl AlwaysAutofixableViolation for RepeatedIsinstanceCalls {
     #[derive_message_formats]
     fn message(&self) -> String {
-        let RepeatedIsinstanceCalls { obj, types } = self;
-        let types = types.join(", ");
-        format!("Merge these isinstance calls: `isinstance({obj}, ({types}))`")
+        let RepeatedIsinstanceCalls { expr } = self;
+        format!("Consider merging these isinstance calls to `{expr}`")
+    }
+
+    fn autofix_title(&self) -> String {
+        let RepeatedIsinstanceCalls { expr } = self;
+        format!("Replace with `{expr}`")
+    }
+}
+
+fn merged_isinstance_call(
+    obj: &str,
+    types: impl IntoIterator<Item = String>,
+    target_version: PythonVersion,
+) -> String {
+    if target_version >= PythonVersion::Py310 {
+        format!("isinstance({}, {})", obj, types.into_iter().join(" | "))
+    } else {
+        format!("isinstance({}, ({}))", obj, types.into_iter().join(", "))
```

Comment:
Should we use the AST to build this or is it fine to format it directly?

Reply:
I think either would be reasonable.